### ISSUE TITLE
Typecheck fails to operate on generics in certain circumstances

### ISSUE THREAD

**Issue by denisvmedia** [golangci/golangci-lint#5832]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Official GitHub Action

### Description of the problem

https://github.com/denisvmedia/inventario/actions/runs/15247011539/job/42875453229

```
  Error: /home/runner/work/inventario/inventario/go/apiserver/areas_test.go:25:87: not enough arguments in call to must.Must
  	have (unknown type)
  	want (T, error) (typecheck)
  	expectedAreas := must.Must(params.RegistrySet.AreaRegistry.List(context.Background()))
  	                                                                            ^
```

It uses https://github.com/go-extras/go-kit/blob/master/must/must.go. List returns 2 values and must.Must expects 2 args. But for some reason typecheck fails to understand this.

P.S. On windows it works differently and actually detects real linting issues.

### Version of golangci-lint

<details>

```console
Run golangci/golangci-lint-action@v8
  with:
    args: --timeout=30m
    version: v2.1.6
    working-directory: go
    install-mode: binary
    github-token: ***
    verify: true
    only-new-issues: false
    skip-cache: false
    skip-save-cache: false
    problem-matchers: false
    cache-invalidation-interval: 7
```

</details>

### Configuration

<details>

```console
Run golangci/golangci-lint-action@v8
  with:
    args: --timeout=30m
    version: v2.1.6
    working-directory: go
    install-mode: binary
    github-token: ***
    verify: true
    only-new-issues: false
    skip-cache: false
    skip-save-cache: false
    problem-matchers: false
    cache-invalidation-interval: 7
```

</details>

### Go environment

<details>

```console
Run actions/setup-go@v5
  with:
    go-version: 1.24.1
    cache: true
    check-latest: false
```

</details>

### Verbose output of running

<details>

The following is not applicable, I'm in CI.

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

https://github.com/denisvmedia/inventario

```go
  Error: /home/runner/work/inventario/inventario/go/apiserver/areas_test.go:25:87: not enough arguments in call to must.Must
  	have (unknown type)
  	want (T, error) (typecheck)
  	expectedAreas := must.Must(params.RegistrySet.AreaRegistry.List(context.Background()))
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by denisvmedia** [comment]
Well, it does compile, when dist is present (it is present, when I run tests for example). I can make sure that it is present when I run lint rules. However the problem is that these are false alarms. Is there a way to improve that?
